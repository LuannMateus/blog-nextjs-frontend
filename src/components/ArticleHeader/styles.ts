import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    max-width: 100%;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
