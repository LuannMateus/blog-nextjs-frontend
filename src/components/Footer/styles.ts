import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    width: 100%;

    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
