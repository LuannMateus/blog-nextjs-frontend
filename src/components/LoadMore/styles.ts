import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    width: 16rem;

    border: 1px solid ${theme.colors.darkText};

    background: ${theme.colors.primary};
    color: ${theme.colors.white};

    display: flex;
    justify-content: center;
    align-items: center;

    padding: ${theme.spacings.small};

    cursor: pointer;

    transition: filter 300ms ease-in-out;

    &:hover {
      filter: brightness(50%);
    }

    &:disabled {
      opacity: 0.4;
    }
  `}
`;
