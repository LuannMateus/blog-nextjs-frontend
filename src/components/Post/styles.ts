import styled, { css } from 'styled-components';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;

    padding-left: ${theme.spacings.small};
  `}
`;
