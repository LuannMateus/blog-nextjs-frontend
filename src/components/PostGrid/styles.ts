import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 100%;
    width: 100%;

    margin: 0 auto;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};

    padding: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      padding: 0;
    }
  `}
`;
