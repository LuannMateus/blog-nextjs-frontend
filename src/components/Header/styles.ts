import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    max-width: ${theme.sizes.max};
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    padding-left: ${theme.spacings.small};
    padding-bottom: ${theme.spacings.xhuge};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin-right: calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.lteMedium} {
      margin: 0;

      flex-direction: column;
    }
  `}
`;

export const Content = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.medium};

    @media ${theme.media.lteMedium} {
      text-align: center;

      margin-top: ${theme.spacings.small};
    }
  `}
`;
