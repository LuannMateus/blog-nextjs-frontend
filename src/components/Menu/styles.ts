import styled, { css } from 'styled-components';
import { Container as LogoLinkStyles } from '../LogoLink/styles';

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    width: 27rem;
    height: 100vh;

    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #000;
    border-radius: 4px;

    padding-top: 1.4rem;

    position: fixed;
    left: -26rem;

    transition: all 300ms ease-in-out;

    z-index: 1000;

    .menuIcon {
      position: relative;
      left: 1.5rem;

      background: #000;
      padding: 0;
    }

    svg {
      width: 3rem;
      color: ${theme.colors.secondary};

      align-self: flex-end;

      padding-right: 1.6rem;

      cursor: pointer;
      transition: filter 300ms ease-in-out;
    }

    svg:hover {
      filter: brightness(50%);
    }

    ${LogoLinkStyles} {
      margin-top: ${theme.spacings.xhuge};
    }

    &.active {
      left: 0rem;
    }
  `}
`;

export const MenuLinkContainer = styled.section`
  ${() => css`
    width: 90%;

    margin-top: 4rem;
  `}
`;
