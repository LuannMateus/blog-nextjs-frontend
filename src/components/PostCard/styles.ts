import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    img {
      transition: filter 300ms ease-in-out;
      width: 100%;
      max-width: 100%;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      filter: brightness(80%);
    }
  `};
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
`;

export const Excerpt = styled.p``;
