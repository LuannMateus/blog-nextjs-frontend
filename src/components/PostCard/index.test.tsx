import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { PostCard } from '.';

import props from './mock';

describe('<PostCard />', () => {
  it('Should render a heading, cover and excerpt', () => {
    renderTheme(<PostCard {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: props.title })).toBeInTheDocument();
    expect(screen.getByText(props.excerpt)).toBeInTheDocument();
    expect(
      screen.getAllByRole('link', { name: props.title })[0],
    ).toHaveAttribute('href', `/posts/${props.slug}`);
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
