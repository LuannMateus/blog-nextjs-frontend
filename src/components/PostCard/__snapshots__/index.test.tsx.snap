// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
.c3 {
  line-height: 1.2;
  color: #000000;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 .c2 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: 1.6rem;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 img {
  -webkit-transition: filter 300ms ease-in-out;
  transition: filter 300ms ease-in-out;
  width: 100%;
  max-width: 100%;
}

.c0:hover a {
  color: #dc143c;
}

.c0:hover img {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.c1 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <section
    class="c0"
  >
    <a
      href="/posts/usando-effect-hook-hook-de-efeito"
    >
      <img
        alt="Usando Effect Hook (Hook de Efeito)"
        class="c1"
        src="https://res.cloudinary.com/project-cdn/image/upload/v1643992684/use_Effect_hook_62405844a5.png"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a
        href="/posts/usando-effect-hook-hook-de-efeito"
      >
        Usando Effect Hook (Hook de Efeito)
      </a>
    </h2>
    <p
      class=""
    >
      O Effect Hook (Hook de Efeito) te permite executar efeitos colaterais em componentes funcionais.
    </p>
  </section>
</div>
`;
